### PR TITLE
feat: add support for us govcloud with AWS federated roles

### PR DESCRIPTION
**Pull request by mikedizon**
**Changelog**

* Added support for US GovCloud
* Added support for opening a web console with US GovCloud

**Bugfixes**

* Added the IdP URL for Azure AD
* Added GovCloud SAML URL

### CODE REVIEW

File: packages/core/src/services/aws-saml-assertion-extraction-service.ts [hunk 1]
```diff
@@ -14,10 +14,14 @@ const authenticationUrlRegexes = new Map([
       /^https:\/\/login\.okta\.com\/.*/,
       /^https:\/\/accounts\.google\.com\/ServiceLogin.*/,
       /^https:\/\/login\.microsoftonline\.com\/.*\/oauth2\/authorize.*/,
+      /^https:\/\/account\.activedirectory\.windowsazure\.com\/applications\/signin\/*?tenantId=*/,
     ],
   ],
 ]);
-const samlAssertionRegexes = new Map([[CloudProviderType.aws, [/^https:\/\/signin\.aws\.amazon\.com\/saml/]]]);
```
Comment (mholttech) [inline packages/core/src/services/aws-saml-assertion-extraction-service.ts:20]: I would suggest swapping this one as well to have the public AWS Cloud first and GovCloud Second

File: packages/core/src/services/aws-saml-assertion-extraction-service.ts [hunk 2]
```diff
@@ -14,10 +14,14 @@ const authenticationUrlRegexes = new Map([
...
+
+const samlAssertionRegexes = new Map([
+  [CloudProviderType.aws, [/^https:\/\/signin\.amazonaws-us-gov\.com\/saml/, /^https:\/\/signin\.aws\.amazon\.com\/saml/]],
+]);
```
Comment (mholttech) [inline packages/core/src/services/aws-saml-assertion-extraction-service.ts:24]: I would suggest swapping this one as well to have the public AWS Cloud first and GovCloud Second

File: packages/core/src/services/aws-saml-assertion-extraction-service.ts [hunk 3]
```diff
@@ -14,10 +14,14 @@ const authenticationUrlRegexes = new Map([
       /^https:\/\/login\.okta\.com\/.*/,
       /^https:\/\/accounts\.google\.com\/ServiceLogin.*/,
       /^https:\/\/login\.microsoftonline\.com\/.*\/oauth2\/authorize.*/,
+      /^https:\/\/account\.activedirectory\.windowsazure\.com\/applications\/signin\/*?tenantId=*/,
```
Comment (mholttech) [inline packages/core/src/services/aws-saml-assertion-extraction-service.ts:17]: Adding account.activedirectory.windowsazure.com may not be necessary. If I recall from when I was working through this with the Noovolari team a while back this URL redirects a few times and lands on login.microsoftonline.com, which is where the authentication process begins.

If I recall from our Slack conversation you were getting the authentication window without this but were landing on the AWS Console Page due to the missing govcloud regex that you added below.

Comment (mikedizon) [reply packages/core/src/services/aws-saml-assertion-extraction-service.ts:17]: Thanks! I removed the regex and it works fine.

File: packages/core/src/services/aws-saml-assertion-extraction-service.ts (hunk 2 above)
Comment (andreacavagna01) [reply packages/core/src/services/aws-saml-assertion-extraction-service.ts]: I think it's something that can be made after the merge, thanks to both for the contribution!

**Comment by andreacavagna01** [comment]
Thanks for the contribution! I think it's been reviewed and can be merged

**Pull request merged**